plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3Service.java

```diff
@@ -208,6 +242,20 @@ static ClientConfiguration buildConfiguration(S3ClientSettings clientSettings) {
         return clientConfiguration;
     }
 
+    private static SSLConnectionSocketFactory createSocksSslConnectionSocketFactory(final InetSocketAddress address) {
```

Comment:
this is exactly the same as AWS SDK creates `SSLConnectionSocketFactory`.  Since the plugin does not use additional  settings which could affect its settings. IMHO it is more or less ok.